File: MvvmJavaDemo/app/src/main/java/com/android/mvvmjava/view/ui/base/CoreActivity.java
package com.android.mvvmjava.view.ui.base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.android.mvvmjava.MvvmApp;

abstract public class CoreActivity extends AppCompatActivity {

    protected MvvmApp mvvmApp;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mvvmApp = (MvvmApp) getApplication();
    }
}

File: BetterStetho/app/src/main/java/com/android/betterstetho/CoreActivity.kt
package com.android.betterstetho

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

abstract class CoreActivity : AppCompatActivity() {

    protected lateinit var coreApp: CoreApp
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        coreApp = application as CoreApp
    }
}